### PR TITLE
controller: Introducing EventProcessor interface - provides an event actor

### PR DESCRIPTION
**Pull request by draychev**
@asridharan Following up on [your comment](https://github.com/Azure/application-gateway-kubernetes-ingress/pull/149#discussion_r281185114) on https://github.com/Azure/application-gateway-kubernetes-ingress/pull/149 and my promise to introduce this Interface.

Here it is - a bit verbose, but otherwise unambigous interface: `QueuedEventEventProcessor`

### CODE REVIEW

File: pkg/controller/eventqueue.go [hunk 1]
```diff
@@ -6,40 +6,47 @@
...
 	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/util/workqueue"
 
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/utils"
 )
 
+// QueuedEventEventProcessor provides a mechanism to act on events in the internal queue.
+type QueuedEventEventProcessor interface {
```
Comment (asridharan) [inline pkg/controller/eventqueue.go]: Why not just `EventProcessor`?This is pretty verbose.

Comment (draychev) [reply pkg/controller/eventqueue.go]: Oh wow - this indeed got out of hand.

**Comment by draychev** [comment]
@asridharan PTAL

**Comment by draychev** [comment]
Fixed comment-lint error *after* the last LGTM. Merging.

**Pull request merged**